data_manager_select_index_by_path/data_manager/path_name_value_key_manager.py

```diff
@@ -1,104 +1,134 @@
+    parser.add_argument('--name', action='store', type=str, default=None, help='name')
+    parser.add_argument('--path', action='store', type=str, default=None, help='path', required=True)
+    parser.add_argument('--data_table_name', action='store', type=str, default=None, help='Name of the data table',
+                        required=True)
+    parser.add_argument('--json_output_file', action='store', type=Path, default=None, help='Json output file',
+                        required=True)
+    return parser
 
-def check_param(name, value, default=None,  check_tab=True):
-    if value in [ None, '', '?' ]:
-        if default:
-            print "Using {0} for {1} as no value provided".format( default, name )
-            value = default
-        else:
-            raise Exception( '{0} is not a valid {1}. You must specify a valid {1}.'.format( value, name ) )
-    if check_tab and "\t" in value:
-        raise Exception( '{0} is not a valid {1}. It may not contain a tab because these are used as seperators by galaxy .'.format( value, name ) )
-    return value
 
-def prefix_exists(directory, prefix):
-    '''checks if files exist with prefix in a directory. Returns Boolean'''
-    matched_files = []
-    directory_files = os.listdir(directory)
-    for directory_file in directory_files:
-        if directory_file.startswith(prefix):
-            matched_files.append(directory_file)
-    # Empty list should return False
-    return bool(matched_files)
+def check_tab(name: str, value: str):
+    if '\t' in value:
+        raise ValueError(
+            '{0} is not a valid {1}. It may not contain a tab because these are used as seperators by galaxy .'.format(
+                value, name))
+
 
-def prefix_plus_extension_exists(directory, prefix, extension):
-    '''checks if files exist with prefix in a directory. Returns Boolean'''
+def prefix_plus_extension_exists(directory: Path, prefix: str, extension: str):
+    """checks if files exist with prefix in a directory. Returns Boolean"""
     matched_files = []
-    directory_files = os.listdir(directory)
-    for directory_file in directory_files:
-        if directory_file.startswith(prefix) and directory_file.endswith(extension):
+    for directory_file in directory.iterdir():
+        if directory_file.name.startswith(prefix) and directory_file.name.endswith(extension):
             matched_files.append(directory_file)
     # Empty list should return False
     return bool(matched_files)
 
+
+class DataTable(object):
+
+    def __init__(self,
+                 index_path: Path,
+                 data_table_name: str,
+                 indexes_properties_file: Path,
+                 name=None,
+                 dbkey=None,
+                 value=None,
+                 ):
+        self.index_path = index_path
+        self.data_table_name = data_table_name
+        self.name = name if name else self.index_path.with_suffix('').name.__str__()
```

Comment:
Is there a reason why the special `__str__` method is called as opposed to simply `str(self.index_path.with_suffix('').name`)?